app/screens/edit_server/index.tsx

```diff
@@ -57,7 +63,20 @@ const EditServer = ({closeButtonId, componentId, server, theme}: ServerProps) =>
     }, [componentId]);
 
     useEffect(() => {
-        setButtonDisabled(Boolean(!displayName || displayName === server.displayName));
+        const loadCredentials = async () => {
+            const credentials = await getServerCredentials(server.url);
+            const hasPassword = Boolean(credentials?.preauthSecret);
+
+            // If there's an existing password, show dummy value "keep"
+            const initialValue = hasPassword ? 'keep' : '';
```

Comment:
We should get feedback from UX on this. It feels weird that in order to change it, I have to remove a complete word.

Nevertheless, if we want to continue this path, I guess this string probably should be translated.